Add 2022-annual-review docs (#13)
karmada-io/community: cncf_process/2022_karmada_annual/2022-karmada-annual.md

In the past few releases, contributors to our project are steadily growing and the activity of our project continues to rise refer to the [Contribution chart](https://karmada.devstats.cncf.io/d/74/contributions-chart?orgId=1&from=now-1y&to=now&var-period=d7&var-metric=contributions&var-repogroup_name=All&var-country_name=All&var-company_name=All&var-company=all).
At the same time, a total of 182 contributors came from 50 different companies and organizations, showing the diversity of the community.
Among them, there are members from cloud vendors, members from academic institutions, and some independent developers.

![Contributors](./images/contributors.png)

Among the contributors, there are **70** members which means they are continuously contributing to our project.
Many of these people have been recognized by the community as reviewers and approvers, and they still demonstrate the diversity and inclusion of the community.
Listed below are some of the community reviewers and approvers. Most of them are new contributors of our projects last year.
We are very excited to see them grow in the community and look forward to them taking on greater responsibilities.

> | Reviewer/Approver | GitHub ID        | Affiliation             | Email                                                                             |

Author: Add complete lists.

| ----------------- | ---------------- | ----------------------- | --------------------------------------------------------------------------------- |

> | Zhe Cheng         | @lonelyCZ        | SEL-Zhejiang University | [531187475@qq.com](mailto:531187475@qq.com)                                       |

Reviewer: I have already applied for a Campus Email and may be able to use it.

```suggestion
| Zhe Cheng         | @lonelyCZ        | SEL-Zhejiang University | [chengzhe@zju.edu.cn](mailto:chengzhe@zju.edu.cn)                                       |
```

including holding meetups and online communication activities. Also, we
continue to refine our documentation and use cases so that our users can get started with our project faster.

Before the Karmada project was born, we have been deeply involved in the Kubefed project led by the Kubernetes multi-cluster sig.
Arguably, Karmada is developed in continuation of Kubernetes Federation v1 and v2 and also their natural successor. Some basic concepts are inherited from these two versions.
In the past year, we continue to collaborate with upstream communities.

- We communicate closely with the Kubernetes community. We participated in and led the design of the [work api](https://github.com/kubernetes-sigs/work-api) in the multi-cluster aspect. Contribution to this project also makes us the [maintainers of this project](https://github.com/kubernetes-sigs/work-api/blob/master/OWNERS).
  Meanwhile, we are actively involved in the [mcs-api](https://github.com/kubernetes-sigs/mcs-api) and [about-api](https://github.com/kubernetes-sigs/about-api) projects hosted by Kubernetes multi-cluster sig.

>   We are also one of the earliest projects to apply the above api of the upstream community. Our users can easily use them in Karmada. For example, we provide [multi-cluster service discovery](https://karmada.io/docs/next/userguide/service/multi-cluster-service) which is based on Multi-cluster Service APIs.

Reviewer 2: ```suggestion
  We are also one of the earliest projects to adopt the above APIs of the upstream communities. Our users can easily use them in Karmada. For example, we provide [multi-cluster service discovery](https://karmada.io/docs/next/userguide/service/multi-cluster-service) based on multi-cluster service APIs.
```

> - Besides Kubernetes, we are open to intergration of more CNCF projects and already have some amazing integration cases. For specific integration cases, refer to [here](#rich-integration).

Reviewer 2: ```suggestion
- Besides Kubernetes, we are open to intergration of more CNCF projects and already have some amazing cases (details available [here](#rich-integration)).
```

> Through the above efforts, the community has grown since the project entered the CNCF sandbox.

Reviewer 2: ```suggestion
The above efforts take the community to a new height since the project entered the CNCF sandbox.
```

From the data we can clearly see this trend.

Karmada currently has [330+](https://karmada.devstats.cncf.io/d/18/overall-project-statistics-table?orgId=1&viewPanel=1) contributors from [50+](https://karmada.devstats.cncf.io/d/5/companies-table?orgId=1) companies and [14](https://karmada.devstats.cncf.io/d/18/overall-project-statistics-table?orgId=1&viewPanel=1) countries.

> For specific developer data, refer to [Diverse developer ecosystem](#diverse-developer-ecosystem).

Reviewer 2: ```suggestion
More details about our developers are available in [Diversifed Developer Ecosystem](#diversified-developer-ecosystem).
```

  What's more, we introduced `karmada-descheduler` to reschedule in case of cluster resource imbalance. Based on this, users can deliver their applications more easily among clusters across regions and vendors.
* **Easier to manage the cluster and the resources on the cluster**. In the past year, we introduced a global uniform resource view across clusters.
  - we enhance the framework of the resource interpreter to enable users to customize the resource status collected to the control plane.
  - We introduced the aggregated API to allow users to connect and operate member clusters from the control plane.
  - We introduced a new component `karmada-search` to implement global search and global proxy for multi-cluster resources.
    Based on this, users no longer have to deal with tedious and repetitive configuration management, and no longer have to endure the management costs caused by fragmented API access portals.

> * **More convenient cross-cluster service governance**. In the past year, we support multi-cluster service discovery and an one-stop multi-cluster network solution. Now users can enable service governance for cross-cluster with Karmada.

Reviewer 2: ```suggestion
* **More convenient cross-cluster service governance**. We support multi-cluster service discovery and a one-stop multi-cluster network solution. Now users can enable service governance for cross-cluster with Karmada.
```

> * **High availability for multi-cloud applications**. In the past year, we support cross-cluster failover of applications.

Reviewer 2: ```suggestion
* **High availability for multi-cloud applications**. We support cross-cluster failover of applications.
```

>   When a cluster fails, applications can smoothly migrate from the failed cluster to another available cluster. During the replicas migration process, Karmada can ensure that the service replicas does not drop to zero, thereby ensuring that the service will not be interrupted.

Reviewer 2: ```suggestion
When a cluster fails, applications can smoothly migrate from the failed cluster to another available cluster. During the replicas migration, Karmada ensures that the service replicas do not drop to zero, thereby ensuring that the service will not be interrupted.
```

  Based on this, users can use Karmada to manage distributed applications with confidence.
* **Smoother multi-cluster O&M experience**. In the past year, we continue to optimize our command line tools. Now users can complete the creation of the Karmada control plane and bootstrap token-based cluster registration with one click.

>   Meanwhile, we provide not only a smooth upgrading path from the previous version but also a clear approach to migrate from Kubefed or promote legacy workload from Kubernetes.

Reviewer 2: ```suggestion
Meanwhile, we provide not only a smooth version upgrade path but also a clear approach to migrate from Kubefed or promote legacy workloads from Kubernetes.
```

> What are the current goals of the project? For example, are you working on major new features? Or are you concentrating on adoption or documentation?

> The Karmada project continues to grow and has its sight on multiple goals for both the near term and long range:

Reviewer 2: ```suggestion
The Karmada project continues to grow and has its sight on short-term and long-term goals:
```

* **Multi-cluster security**. Multi-cluster security presents many challenges.
  - Karmada manages the cluster on the central side, which simplifies the tedious steps of cluster management but also brings some security risks. 

>     We should minimize the permissions of the control plane and reduce blast radius to member clusters after a control plane failure.

Reviewer 2: ```suggestion
The aim is to minimize the permissions of the control plane and reduce blast radius to member clusters after a control plane failure.
```

  - Meanwhile, multi-cluster access control with high fine-grained security isolation on the control plane is also a major direction in the future.
* **Performance to adapt to larger scale landing scenarios**. We have started an early effort around improving user experiences in large scale scenarios in the past releases.
  Multiple cloud naturally means larger resource pools, which brings greater performance pressure to the multi-cloud management platform.
  In large-scale scenarios, we will continue to reduce the access delay, save the resources consumed, and enhance the stability of the system under high concurrency conditions.
* **Multi-cluster observability**. Multi-cluster observability is an important part of a multi-cluster orchestration.
  - We will continue to explore integrated solutions for monitoring and logging at the multi-cloud era.
  - We will continue to optimize multi-cluster events so that users can better operate and maintain the multi-cloud management platform.
* **Multi-cluster application distribution**. There are still many challenges in application distribution across multiple clusters.

>   - When the plug-in is coupled with the cluster, how to simplify the installation of the plug-in in the target cluster and manage the lifecycle of the plug-in is our next major research direction.

Reviewer 2: ```suggestion
  - For plug-ins coupled with the cluster, we will dive into simplifying the plug-in installation and lifecycle management.
```

>   - When applications have sequential relationships between clusters, we should provide a multi-cluster workflow to simplify the deployment of these applications.

Reviewer 2: ```suggestion
  - When applications have sequential relationships between clusters, we aim to provide a multi-cluster workflow to simplify the deployment of these applications.
```

* **Integration with ecosystem**.
  The integration with the CNCF ecosystem has always been a major focus of our efforts.
  We believe that cooperation in different fields can better serve our users.
  We will continue to seek cooperation with other mature projects to form an open and prosperous ecosystem.

> The planed new features are shown in [roadmap](https://github.com/karmada-io/karmada/blob/master/ROADMAP.md).

Reviewer 2: ```suggestion
The planned new features are listed in [roadmap](https://github.com/karmada-io/karmada/blob/master/ROADMAP.md).
```

## How the CNCF Can Help

Similar to many other Sandbox projects, we wish to get the following help from CNCF:

> - More channels to advocate the project.

Reviewer 2: ```suggestion
- More channels for exposure
```

> - More chances to collaborate with other projects in CNCF or even out of CNCF.

Reviewer 2: ```suggestion
- More chances to collaborate with other projects, both CNCF or non-CNCF ones.
```

> - Technical writing support for project documents.

Reviewer 2: ```suggestion
- Technical writing support
```

are not aware of cluster differences. 4) Duplicate configuration. Although services are configured on the cloud management platform in a unified manner, their configurations need to be delivered to each cluster separately.
To solve these problems, ICBC finally chooses Karmada. Shen Yifan at ICBC said:

  > "For cross-cluster deployment models and scheduling, Karmada provides independent propagation policies, defined using a policy API, that can be reused across applications. For example, ICBC has its independent platform team and service team. This model can well adapt to their organizational structures."
  >

  The use case details can be found [here](https://karmada.io/zh/blog/2022/03/22/multi-cluster-management-with-an-ocean-of-nodes/).
* **Vivo** is a tech company that designs and develops smartphones, smartphone accessories, software and online services. To better manage a large-scale cloud native infrastructure, Vivo has the following requirements: 1) It has the ability to manage multiple Kubernetes clusters in a unified manner, and the application manages resources through the service dimension. 2) It has cross-cluster elastic scaling and scheduling capabilities to

>   achieve rational utilization of cross-cluster resources. 3) Dynamic migration of applications to guarantee quick recovery in case of failure. 4) It has the ability to improve the disaster.

Reviewer 2: ```suggestion
  more efficient use of cross-cluster resources; 3) Dynamic migration of applications for quick failover; 4) Better DR
```

>   To solve the above problem, Vivo finally chooses Karmada. In the words of Zhang Rong at Vivo:

Reviewer 2: ```suggestion
  To obtain the above capabilities, Vivo chooses Karmada. According to Zhang Rong from Vivo:
```

  > TBD
  >

> * **Trip.com** is a Chinese multinational online travel company that provides services including accommodation reservation, transportation ticketing, packaged tours and corporate travel management. Before using Karmada, Trip.com used KubeFed to build a multi-cloud container platform. However, KubeFed uses non-native Kubernetes APIs, which makes it difficult to migrate existing clusters. In addition, its community is becoming less active.

Reviewer 2: ```suggestion
* **Trip.com** is a Chinese multinational online travel company that provides services including accommodation reservation, transportation ticketing, packaged tours and corporate travel management. Before using Karmada, Trip.com used KubeFed to build their multi-cloud container platform. However, KubeFed uses non-native Kubernetes APIs, which makes it difficult to migrate existing clusters to their multi-cloud platform. The KubeFed community is also becoming inactive.
```

>   So Trip.com seeks a multi-cloud container platform with smooth migration from KubeFed. Besides, it hopes that the platform can have cross-cluster elastic scaling and scheduling capabilities. In the words of Yue HongHui at Trip.com:

Reviewer 2: ```suggestion
  So Trip.com seeks smooth migration from KubeFed, as well as cross-cluster auto scaling and scheduling for their multi-cloud platform. Yue HongHui from Trip.com said:
```
